lightningnetwork/lnd
# Add possibility to define a change scope in FundPsbt

## Change Description

Currently in LND, we use taproot addresses for change outputs by default in FundPsbt:
https://github.com/lightningnetwork/lnd/pull/7193

But, a lot of users might be impacted by this change after upgrading their nodes to a newer LND version (because it seems that it is a breaking change).

This PR aims to solve this problem and makes possible to define a custom change address in FundPsbt.
Now FundPsbt defaults to P2WKH for change address for default/imported account

This PR uses a previous PR to choose a custom scope in btcwallet.FundPsbt:
https://github.com/btcsuite/btcwallet/pull/842 

This PR gives this possibility in lncli and in walletrpc

## Steps to Test

I've tested this PR in regtest using a custom node pointing to this branch using the CLI and grpc calls

To test, you can simply build a lnd image using this branch and check that:

- `lncli wallet psbt fund` provides the possibility to modify the change scope using `--change-address`
- check that the FundPsbt grpc call provides the optional `change-address` argument

## Pull Request Checklist
### Testing

### Code Style and Documentation

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.
